### PR TITLE
Remove autotools build

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
We use GN as the primary build system.
Autotools is hard to maintain and update and GN is assumed to have achieved parity.

 #### Summary of Changes
Removes autotools from CI and removes some autotools build files. We may need additional passes as we find more legacy autotools bit, but this is the first large CL.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: lgtm.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ extraction:
         configure: # Customizable step used only by C/C++ extraction.
             command:
                 - source scripts/bootstrap.sh && gn gen out/custom
-                  --args='target_os="linux_x64"'
+                  --args='target_os="linux"'
```
Comment (mspang) [inline lgtm.yml:9]: You can leave target_os blank.

Comment (andy31415) [reply lgtm.yml:9]: Would that make it host_os?
i.e. should I just drop the the --args parameter?

Comment (andy31415) [reply lgtm.yml]: I dropped the args for now.

Comment (mspang) [reply lgtm.yml]: Yes, the default is host_os so dropping args completely is equivalent.

**Review by rwalker-apple — APPROVED** [review]

**Pull request merged**